erts/emulator/internal_doc/beam_makeops.md

```diff
@@ -1874,3 +1974,90 @@ other two only have two fragments.  Here are the fragments:
 
 The full definitions of those instructions can be found in `bs_instrs.tab`.
 The generated code can be found in `beam_warm.h`.
+
+### Code generation for BeamAsm ###
+
+For the BeamAsm runtime system, the implementation of each instruction is defined
+by emitter functions written in C++ that emit the assembly code for each instruction.
+There is one emitter function for each family of specific instructions.
+
+Take for example the `move` instruction. In `beam/asm/ops.tab` there is a
+single specific instruction for `move` defined like this:
+
+    move s d
+
+The implementation is found in `beam/asm/instr_common.cpp`:
+
+    void BeamModuleAssembler::emit_move(const ArgVal &Src, const ArgVal &Dst) {
+        mov_arg(Dst, Src);
+    }
+
+The `mov_arg()` helper function will handle all combinations of source and destination
+operands.  For example, the instruction `{move,{x,1},{y,1}}` will be translated like this:
+
+    mov rdi, qword [rbx+8]
+    mov qword [rsp+8], rdi
+
+while `{move,{integer,42},{x,0}}` will be translated like this:
+
+    mov qword [rbx], 687
+
+It is possible to define more than one specific instruction, but there will still be
+only one emitter function. For example:
+
+    fload S l
+    fload q l
+
+By defining `fload` like this, the source operand must be a X register, Y register, or
+a literal.  If not, the loading will be aborted.  If the instruction instead had been
```

Comment:
what's the impact of aborting loading on the VM? A complete crash?

Reply:
The module won't be loaded, and an error message will be printed out.